src/electron/go_vpn_tunnel.ts

```diff
@@ -241,39 +239,52 @@ class GoTun2socks {
       args.push('-dnsFallback');
     }
 
-    return new Promise<void>((resolve, reject) => {
-      this.process.onExit = (code?: number) => {
-        reject(errors.fromErrorCode(code ?? errors.ErrorCode.UNEXPECTED));
-      };
-      this.process.onStdErr = (data?: string | Buffer) => {
-        if (!data?.toString().includes('tun2socks running')) {
-          return;
-        }
-        console.debug('tun2socks started');
-        this.process.onExit = async (code?: number, signal?: string) => {
-          // The process exited unexpectedly, restart it.
-          console.warn(`tun2socks exited unexpectedly with signal: ${signal}, code: ${code}. Restarting...`);
-          await this.start(isUdpEnabled);
-        };
-        this.process.onStdErr = null;
-        resolve();
-      };
-      this.process.launch(args);
-    });
+    this.autoRestart = true;
+    this.process.onStdErr = async (data?: string | Buffer) => {
+      if (!data?.toString().includes('tun2socks running')) {
```

Comment:
Ugh, good example why we need to use Go as a library or with IPCs.

Reply:
Definitely, it will be much simpler to report errors in a library.